src/BuiltInTools/DotNetDeltaApplier/PipeListener.cs

```diff
@@ -155,8 +171,22 @@ private async ValueTask ReadAndApplyStaticAssetUpdateAsync(
         // Updating static asset only invokes ContentUpdate metadata update handlers.
         // Failures of these handlers are reported to the log and ignored.
         // Therefore, this request always succeeds.
-        var response = new UpdateResponse(logEntries, success: true);
+        await SendResponseAsync(new UpdateResponse(logEntries, success: true), cancellationToken);
+    }
 
-        await response.WriteAsync(pipeClient, cancellationToken);
+    internal async ValueTask SendResponseAsync<T>(T response, CancellationToken cancellationToken)
+        where T : IResponse
+    {
+        Debug.Assert(_pipeClient != null);
+        try
+        {
+            _messageToClientLock.Wait(cancellationToken);
+            await _pipeClient.WriteAsync((byte)response.Type, cancellationToken);
```

Comment:
Is it possible for there to be a race that causes `_pipeClient` to be null here?

Reply:
Good catch. I removed null assignments after disposal, so that the field is always non-null after initialized.